src/main/java/au/org/ala/biocache/util/QueryFormatUtils.java

```diff
@@ -173,7 +189,52 @@ public void setMaxBooleanClauses(int maxBooleanClauses) {
 
         updateQueryContext(searchParams);
 
-        return activeFacetMap;
+        return fqMaps;
+    }
+
+    /**
+     * To split a fq into key and value
+     *
+     * This method assumes fq is in one of below format
+     * fq = key:val or fq = -key:val
+     * fq = (key:val) or fq = (-key:val)
+     * fq = -(key:val) or fq = -(-key:val)
+     * fq = key:val1 OR key:val2 or fq = -key:val1 OR -key:val2
+     * fq = (key:val1 OR key:val2) or fq = (-key:val1 OR -key:val2)
+     * fq = -(key:val1 OR key:val2) or fq = -(-key:val1 OR -key:val2)
+     *
+     * inclusive and exclusive keys can't co-exist in one fq
+     */
+    private String[] parseFQ(String fq) {
```

Comment:
Only the first array element gets used from the result of this method, might be worth returning a single `String`

Reply:
fixed